linera-views/src/common.rs

```diff
@@ -41,6 +41,51 @@ pub(crate) enum Update<T> {
     Set(T),
 }
 
+#[derive(Clone, Debug)]
+pub(crate) struct DeletionSet {
+    pub delete_storage_first: bool,
+    pub deleted_prefixes: BTreeSet<Vec<u8>>,
+}
+
+impl DeletionSet {
+    pub fn new() -> Self {
+        Self {
+            delete_storage_first: false,
+            deleted_prefixes: BTreeSet::new(),
+        }
+    }
+
+    pub fn clear(&mut self) {
+        self.delete_storage_first = true;
+        self.deleted_prefixes.clear();
+    }
+
+    pub fn rollback(&mut self) {
+        self.delete_storage_first = false;
+        self.deleted_prefixes.clear();
+    }
+
+    pub fn contains_key(&self, index: &[u8]) -> bool {
+        if self.delete_storage_first {
+            return true;
+        }
+        contains_key(&self.deleted_prefixes, index)
```

Comment:
Should this be renamed to `contains_prefix_of`?

Reply:
Yes, and it has the added benefit of diminishing namespace collisions.